solver/src/settlement.rs

```diff
@@ -36,21 +38,25 @@ impl Trade {
     }
 
     // The difference between the minimum you were willing to buy/maximum you were willing to sell, and what you ended up buying/selling
-    pub fn surplus(&self, sell_token_price: U256, buy_token_price: U256) -> Option<U256> {
+    pub fn surplus(
+        &self,
+        sell_token_price: &BigRational,
+        buy_token_price: &BigRational,
+    ) -> Option<BigRational> {
```

Comment:
nit: I wonder if this should return a Result instead of an Option so that we can pass the information about which trade is causing problems which would make it easier to debug. Can happen in a separate PR though as it's already an Option in the current codebase.

Reply:
Will do in subsequent PR.